MatrixSDK/Data/Store/MXMemoryStore/MXMemoryStore.m

```diff
@@ -268,32 +268,41 @@ - (MXReceiptData *)getReceiptInRoom:(NSString*)roomId forUserId:(NSString*)userI
 
 - (NSUInteger)localUnreadEventCount:(NSString*)roomId threadId:(NSString *)threadId withTypeIn:(NSArray*)types
 {
-    // @TODO: This method is only logic which could be moved to MXRoom
+    NSArray<MXEvent*> *newEvents = [self newIncomingEventsInRoom:roomId threadId:threadId withTypeIn:types];
+    __block NSUInteger result = 0;
+    // Check whether these unread events have not been redacted.
+    [newEvents enumerateObjectsUsingBlock:^(MXEvent * _Nonnull event, NSUInteger idx, BOOL * _Nonnull stop)
+    {
+        if (!event.isRedactedEvent)
+        {
+            result++;
+        }
+    }];
+    return result;
+}
+
+- (NSArray<MXEvent *> *)newIncomingEventsInRoom:(NSString *)roomId
+                                       threadId:(NSString *)threadId
+                                     withTypeIn:(NSArray<MXEventTypeString> *)types
+{
     MXMemoryRoomStore* store = [self getOrCreateRoomStore:roomId];
     RoomReceiptsStore *receiptsStore = [self getOrCreateRoomReceiptsStore:roomId];
-    NSUInteger count = 0;
-    
+
     if (store && receiptsStore)
```

Comment:
I would negate this check and bail out early to reduce the indentation level. Same with the `data` check below.

Reply:
I'm not sure i'm following you on this one. This is because my Xcode settings `Automatically trim trailing whitespace` and `Including whitespace-only lines` are turned on. I don't want to turn them off actually, if that's what you want.

Reply:
Ah sorry, I only meant that you could change the control flow a little bit to make it easier to follow:
```
- (NSArray<MXEvent *> *)newIncomingEventsInRoom:(NSString *)roomId
                                       threadId:(NSString *)threadId
                                     withTypeIn:(NSArray<MXEventTypeString> *)types
{
    MXMemoryRoomStore* store = [self getOrCreateRoomStore:roomId];
    RoomReceiptsStore *receiptsStore = [self getOrCreateRoomReceiptsStore:roomId];

    if (store == nil  || receiptsStore == nil)
    {
        return @[]
    }

    MXReceiptData* data = [receiptsStore objectForKey:credentials.userId];
    
    if (data == nil)
    {
        return @[]
    }
    
    // Check the current stored events (by ignoring oneself events)
    return [store eventsAfter:data.eventId
                     threadId:threadId
                       except:credentials.userId
                   withTypeIn:[NSSet setWithArray:types]];
}
```

Reply:
Ah totally misunderstood :) Updated now.